packages/hardhat-core/src/internal/hardhat-network/provider/types/Callback.ts

```diff
@@ -1,2 +1 @@
-export type Callback<T = void> = ((error: NonNullable<any>) => void) &
-  ((error: null | undefined, value: T) => void);
+export type Callback<T = void> = (error: any, value: T) => void;
```

Comment:
This is the type change that triggers all the other typescript changes. This makes sense to me, but I have no idea what I'm doing.